src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_ARM64/ARM64ReadyToRunHelperNode.cs

```diff
@@ -94,6 +94,11 @@ protected override void EmitCode(NodeFactory factory, ref ARM64Emitter encoder,
                                 encoder.EmitJNE(factory.HelperEntrypoint(HelperEntrypoint.EnsureClassConstructorRunAndReturnThreadStaticBase));
                                 EmitInlineTLSAccess(factory, ref encoder);
                             }
+
+                            // REVIEW: how to keep a node around?
```

Comment:
Report it in ComputeNonRelocationBasedDependencies https://github.com/dotnet/runtime/blob/a19585ddbe13e3f485fb18d21782c870c4fffe4b/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ReadyToRunHelperNode.cs#L123

Reply:
I will look into using that. 

The tricky part is to not root threadstatic index for every type that has threadstatics, but only for those that are reachable in the code, even though the index itself is not used. 
(technically the index node is “used” - at compile time, and maybe debug time, but the code itself, once emitted, does not not need the index)

Reply:
Although, I think, we can assume that for every type with inlined storage the index will be used, since we build the storage block based on the analysis that we did in the scanner.

Reply:
I made the node that describes the storage block for threadstatics dependent on the index nodes for the types whose storage is inlined in the block.